crates/milli/src/vector/mod.rs

```diff
@@ -642,19 +683,36 @@ impl Embedder {
     #[tracing::instrument(level = "debug", skip_all, target = "search")]
     pub fn embed_search(
         &self,
-        text: String,
+        text: &str,
```

Comment:
Better as, indeed, there is a change the `&str` is in the cache and doesn't need to be copied.